### PR TITLE
Subscription service adjust

### PR DESCRIPTION
**Pull request by NiazBikbaev**

### CODE REVIEW

File: src/main/java/com/epam/magazinestore/entity/Subscription.java [hunk 1]
```diff
@@ -56,4 +58,34 @@
     @JoinColumn(name = "payment_id")
     private Payment payment;
 
+    public static SubscriptionBuilder newBuilder() {
```
Comment (AntonKiselyov) [inline src/main/java/com/epam/magazinestore/entity/Subscription.java]: Please, read this article: https://projectlombok.org/features/Builder and decide whether the Lombok builder is better than own builder

File: src/main/java/com/epam/magazinestore/services/impl/SubscriptionServiceImpl.java [hunk 2]
```diff
@@ -28,44 +30,87 @@
...
-
-        LocalDate publicationDate = magazine.getPublicationDate();
-        LocalDate endDate = publicationDate.plusMonths(duration);
-
-        subscription.setStartDate(publicationDate);
-        subscription.setEndDate(endDate);
-
-        Payment payment = paymentService.registerPayment(subscription);
+    public Subscription subscribe(Account account, Magazine magazine, int duration, int offset) {
+        if (account == null) {
```
Comment (AntonKiselyov) [inline src/main/java/com/epam/magazinestore/services/impl/SubscriptionServiceImpl.java]: Objects.isNull(account)

File: src/main/java/com/epam/magazinestore/services/impl/SubscriptionServiceImpl.java [hunk 3]
```diff
@@ -28,44 +30,87 @@
...
+            throw new ServiceException("Account can't be null");
+        }
+        if (magazine == null) {
```
Comment (AntonKiselyov) [inline src/main/java/com/epam/magazinestore/services/impl/SubscriptionServiceImpl.java]: Objects.isNull()

File: src/main/java/com/epam/magazinestore/entity/Subscription.java (hunk 1 above)
Comment (NiazBikbaev) [reply src/main/java/com/epam/magazinestore/entity/Subscription.java]: Fixed in 0ec80a2

File: src/main/java/com/epam/magazinestore/services/impl/SubscriptionServiceImpl.java (hunk 2 above)
Comment (NiazBikbaev) [reply src/main/java/com/epam/magazinestore/services/impl/SubscriptionServiceImpl.java]: Fixed in bb54958

File: src/main/java/com/epam/magazinestore/services/impl/SubscriptionServiceImpl.java (hunk 3 above)
Comment (NiazBikbaev) [reply src/main/java/com/epam/magazinestore/services/impl/SubscriptionServiceImpl.java]: Fixed in bb54958

File: src/main/java/com/epam/magazinestore/services/impl/SubscriptionServiceImpl.java [hunk 4]
```diff
@@ -31,21 +32,22 @@
     @Override
     @Transactional
     public Subscription subscribe(Account account, Magazine magazine, int duration, int offset) {
-        if (account == null) {
+        if (Objects.isNull(account)) {
```
Comment (AntonKiselyov) [inline src/main/java/com/epam/magazinestore/services/impl/SubscriptionServiceImpl.java]: Add static import for Objects.isNull()

Comment (NiazBikbaev) [reply src/main/java/com/epam/magazinestore/services/impl/SubscriptionServiceImpl.java]: fixed in a362505

File: src/main/java/com/epam/magazinestore/services/impl/SubscriptionServiceImpl.java [hunk 5]
```diff
@@ -78,6 +79,7 @@ public Subscription unsubscribe(Account account, Magazine magazine) {
             if (subscription == null) {
```
Comment (AntonKiselyov) [inline src/main/java/com/epam/magazinestore/services/impl/SubscriptionServiceImpl.java]: isNull

File: src/test/java/com/epam/magazinestore/SubscriptionServiceTests.java [hunk 6]
```diff
@@ -50,6 +51,7 @@ public void before() {
     }
 
     @Test
+    @Rollback
     public void subscribeTest() {
         Account account = accountService.getAccountByUsername("user");
         Magazine magazine = magazineService.getAvailableMagazines(0, 10).get(0);
```
Comment (AntonKiselyov) [inline src/test/java/com/epam/magazinestore/SubscriptionServiceTests.java]: There are many magic numbers in test. Could you name these as private final static constants in this test-class ?

File: src/test/java/com/epam/magazinestore/SubscriptionServiceTests.java [hunk 7]
```diff
@@ -50,6 +51,7 @@ public void before() {
     }
 
     @Test
+    @Rollback
     public void subscribeTest() {
         Account account = accountService.getAccountByUsername("user");
```
Comment (AntonKiselyov) [inline src/test/java/com/epam/magazinestore/SubscriptionServiceTests.java]: "user" as constants

File: src/main/java/com/epam/magazinestore/services/impl/SubscriptionServiceImpl.java (hunk 5 above)
Comment (NiazBikbaev) [reply src/main/java/com/epam/magazinestore/services/impl/SubscriptionServiceImpl.java]: Resolved in 30548d4

File: src/test/java/com/epam/magazinestore/SubscriptionServiceTests.java (hunk 6 above)
Comment (NiazBikbaev) [reply src/test/java/com/epam/magazinestore/SubscriptionServiceTests.java]: Resolved in 4dd7683

File: src/test/java/com/epam/magazinestore/SubscriptionServiceTests.java (hunk 7 above)
Comment (NiazBikbaev) [reply src/test/java/com/epam/magazinestore/SubscriptionServiceTests.java]: Resolved in 4dd7683

**Pull request merged**